core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java

```diff
@@ -233,6 +202,88 @@ private boolean cacheBlockFromRemoteWorker(long blockId, long blockSize,
     }
   }
 
+  /**
+   * AsyncCacheTask is a runnable task that can be considered equal if
+   * the blockId of the request is the same.
+   */
+  @VisibleForTesting
+  class AsyncCacheTask implements Runnable {
+    private final AsyncCacheRequest mRequest;
+
+    /**
+     * Constructor for an AsyncCacheTask.
+     *
+     * @param request an AsyncCacheRequest
+     */
+    AsyncCacheTask(AsyncCacheRequest request) {
+      mRequest = request;
+    }
+
+    @Override
+    public int hashCode() {
+      // Only care about the block id being the same.
+      // Do not care if the source host or port is different.
+      return Objects.hash(mRequest.getBlockId());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this == obj) {
+        return true;
+      }
+      if (!(obj instanceof AsyncCacheTask)) {
+        return false;
+      }
+      AsyncCacheTask that = ((AsyncCacheTask) obj);
+      if (mRequest == that.mRequest) {
+        return true;
+      }
+      if (this.mRequest == null || that.mRequest == null) {
+        return false;
+      }
+      // Only care about the block id being the same.
+      // Do not care if the source host or port is different.
+      return mRequest.getBlockId() == that.mRequest.getBlockId();
+    }
+
+    @Override
+    public void run() {
+      boolean result = false;
+      long blockId = mRequest.getBlockId();
+      long blockLength = mRequest.getLength();
+      try {
+        boolean isSourceLocal = mLocalWorkerHostname.equals(mRequest.getSourceHost());
+        // Check if the block has already been cached on this worker
+        if (mBlockWorker.hasBlockMeta(mRequest.getBlockId())) {
+          ASYNC_CACHE_DUPLICATE_REQUESTS.inc();
```

Comment:
Does our tracking of duplicate requests need to change now that we do deduplication at the queue level?

Reply:
even though we do dedup at the queue level, there could still be blocks cached either through explicit loading? I think this metric is much more meaningful now and reflects real issues if it is unusually high..

Reply:
I was thinking of preserving `ASYNC_CACHE_REQUESTS` = sum of all the other result request types, but I agree that this metric is a lot more useful than the old one. I'm fine either way.